internal/registry/registry.go

```diff
@@ -89,29 +90,10 @@ func (r *registry) GetLayerByDigest(digest string, pullConfig *RepoPullConfig) (
 	return crane.PullLayer(pullConfig.repo+"@"+digest, pullConfig.authOptions...)
 }
 
-func (r *registry) ExtractToolkitRelease(layer v1.Layer) (*DriverToolkitEntry, error) {
-	var found bool
-	dtk := &DriverToolkitEntry{}
-	obj, err := r.getHeaderFromLayer(layer, driverToolkitJSONFile)
-	if err != nil {
-		return nil, fmt.Errorf("failed to find file %s in image layer: %w", driverToolkitJSONFile, err)
-	}
-
-	dtk.KernelFullVersion, found, err = unstructured.NestedString(obj.Object, "KERNEL_VERSION")
-	if !found || err != nil {
-		return nil, fmt.Errorf("failed to get KERNEL_VERSION from %s, found %t: %w", driverToolkitJSONFile, found, err)
-	}
-
-	dtk.RTKernelFullVersion, found, err = unstructured.NestedString(obj.Object, "RT_KERNEL_VERSION")
-	if !found || err != nil {
-		return nil, fmt.Errorf("failed to get RT_KERNEL_VERSION from %s, found %t: %w", driverToolkitJSONFile, found, err)
-	}
-
-	dtk.OSVersion, found, err = unstructured.NestedString(obj.Object, "RHEL_VERSION")
-	if !found || err != nil {
-		return nil, fmt.Errorf("failed to get RHEL_VERSION from %s, found %t: %w", driverToolkitJSONFile, found, err)
-	}
-	return dtk, nil
+func (r *registry) VerifyModuleExists(layer v1.Layer, pathPrefix, kernelVersion, moduleFileName string) bool {
+	fullPath := filepath.Join(pathPrefix, modulesLocationPath, kernelVersion, moduleFileName)
+	_, err := r.getHeaderStreamFromLayer(layer, fullPath)
```

Comment:
Are we going to make use of the io.Reader` in future code?

Reply:
We might, in case we will need to actually access the file data